flight/Modules/OnScreenDisplay/osd_menu.c

```diff
@@ -1204,19 +1221,25 @@ void brainre1_menu(void)
 			break;
 		case FSM_STATE_RE1_IR_IDTRACKMATE:
 			{
+				uint16_t id = data.IRIDTrackmate;
 				int step = held_long ? 100 : 1;
 				if (current_event == FSM_EVENT_RIGHT) {
-					data.IRIDTrackmate += step;
-					if (data.IRIDTrackmate > MAX_ID_TRACKMATE) {
-						data.IRIDTrackmate = 0;
-					}
+					do {
```

Comment:
This seems less good.  Since like 0xx and 1xx are bad, we could have to do 512 + 16 iterations which seems to be getting excessive.